Azure/azure-cli
# [ServiceFabric] Fix #12470 az sf create cluster, fix bugs in update durability and reliability and find vmss correctly through the code given a node type name

**History Notes:**  
* Fix issue #12470: create cluster was failing when --secret-identifier was not provided on vault creation due to breaking change introduced in the new key vault management sdk. And added tests to cover this case.
* Fix get vmss when node type is different from vmss name, now it always checks in the extension node type reference.
* Fix update durability to do the cluster upgrade first and then the vmss upgrade.
* Fix update reliability: it was removing non primary vmss in the cluster resource with patch request.
* Add test for cluster
    - test_cluster_certs
        - create cluster with --secret-identifier
        - add secondary cert
        - remove primary cert
    - test_cluster
        - create cluster without --secret-identifier
        - add and remove nodes
        - add node type
        - update durability and reliability
        - add and remove cluster setting
---

This checklist is used to make sure that common guidelines for a pull request are followed.
